mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -604,6 +604,7 @@ private void doRenameMailbox(Mailbox mailbox, MailboxPath newMailboxPath, Mailbo
         List<Mailbox> mailboxes = Stream.concat(baseMailboxes, delegatedMailboxes)
             .distinct()
             .filter(Throwing.predicate(mailbox -> storeRightManager.hasRight(mailbox, right, session)))
+            .filter(mailboxQuery::matches)
```

Comment:
I agree with the change but i don't understand why we need this filter in the first place 
since `baseMailboxes` and `delegatedMailboxes` are fetched using this query

Reply:
Because some mappers fails at returning the exact list and return an approximate (as far as I remember JPA & maildir are doing this)